third_party/move/move-model/bytecode/src/stackless_bytecode.rs

```diff
@@ -484,7 +484,37 @@ impl Bytecode {
         self.is_conditional_branch() || self.is_unconditional_branch()
     }
 
-    /// Return the destinations of the instruction
+    /// Return the sources of the instruction.
+    /// If the instruction is a spec-only instruction, return no sources.
+    pub fn sources(&self) -> Vec<TempIndex> {
+        match self {
+            Bytecode::Assign(_, _, src, _) => {
+                vec![*src]
+            },
+            Bytecode::Call(_, _, _, srcs, _) => srcs.clone(),
+            Bytecode::Ret(_, srcs) => srcs.clone(),
+            Bytecode::Branch(_, _, _, cond) => {
+                vec![*cond]
+            },
+            Bytecode::Abort(_, src) => {
+                vec![*src]
+            },
+            Bytecode::Load(_, _, _)
+            | Bytecode::Jump(_, _)
+            | Bytecode::Label(_, _)
+            | Bytecode::Nop(_) => {
+                vec![]
+            },
+            // Note that for all spec-only instructions, we currently return no sources.
+            Bytecode::SaveMem(_, _, _)
+            | Bytecode::SaveSpecVar(_, _, _)
+            | Bytecode::Prop(_, _, _) => {
+                vec![]
```

Comment:
Have not seen last time. Thats kind of wrong semantics, so it would be more adequate to panic that this not implemented.

Reply:
Also, those operations really do not appear in the compiler. They are all introduced as part of code instrumentation in the prover pipeline.

Reply:
Done.